README.md

```diff
@@ -130,6 +247,117 @@ ChatComponent(
 )
 ```
 
+### **Renderers (Graphs, Tables, Attachments & Text)**
+`dash-chat` supports rich content rendering by allowing messages to contain structured content types like graphs, tables, and images. You can render custom content by passing a structured list to the content field of a message.
+
+#### Supported Built-in Renderers
+#### Text
+```python
+{
+    "role": "assistant",
+    "content": {
+        "type": "text",
+        "text": "This will be rendered as a markdown message"
+    },
+}
+```
+
+#### Attachments (Images & Files)
+```python
+{
+    "role": "assistant",
+    "content": {
+        "type": "attachment",
+        "file": "data:image/png;base64,...",
+        "fileName": "example.png",
+        "fileType": "image/png"
+    }
+}
+```
+Renders an image or a downloadable file preview.
+
+#### Graph
+```python
+{
+    "role": "assistant",
+    "content": {
+        "type": "graph",
+        "props": {
+            "figure": {
+                "data": [
+                    {
+                        "x": [1, 2, 3],
+                        "y": [4, 1, 2],
+                        "type": "bar", "name": "Demo"
+                    }
+                ],
+                "layout": {"title": "Bar Chart"},
+            },
+            "config": {"displaylogo": True},
+            "responsive": True
+        }
+    }
+}
+```
+Renders an interactive Plotly graph equivalent to [`dcc.Graph`](https://dash.plotly.com/dash-core-components/graph). The props object supports most of the arguments you would pass to a [`dcc.Graph`](https://dash.plotly.com/dash-core-components/graph).
+
+#### Table
+```python
+{
+    "role": "assistant",
+    "content": {
+        "type": "table",
+        "header": ["Order ID", "Item", "Quantity", "Total"],
+        "data": [
+            ["#1021", "Apple iPhone", 1, "$799"],
+            ["#1022", "Samsung Galaxy", 2, "$1398"]
+        ]
+        "props": {
+            "striped": True,
+            "bordered": True,
+            "hover": True,
+            "responsive": True,
+            "size": "lg"
+        }
+    }
+}
+```
+Renders an HTML table. The props object supports all the arguments you would pass to [`dbc.Table`](https://dash-bootstrap-components.opensource.faculty.ai/docs/components/table/) in dash-bootstrap-components.
```

Comment:
Add a sentence about how to provide the table here, using the keys "header" (a list of strings) and "data" (a list of lists of strings).